beep/structure.py

```diff
@@ -292,18 +318,22 @@ def get_diagnostic_summary(self, diagnostic_available):
             "discharge_energy": "max",
             "charge_energy": "max",
             "temperature": ["max", "mean", "min"],
-            "date_time_iso": "first"})
+            "date_time_iso": "first",
+            "cycle_index": "first"},
+        )
 
         diag_summary.columns = ['discharge_capacity', 'charge_capacity',
                                 'discharge_energy', 'charge_energy',
                                 'temperature_maximum', 'temperature_average',
-                                'temperature_minimum', 'date_time_iso']
+                                'temperature_minimum', 'date_time_iso',
+                                'cycle_index']
         diag_summary = diag_summary[diag_summary.index.isin(diag_cycles_at)]
 
         diag_summary['coulombic_efficiency'] = diag_summary['discharge_capacity'] \
                                                / diag_summary['charge_capacity']
-        diag_summary['diagnostic_type'] = list(itertools.chain.from_iterable(
-            [diagnostic_available['cycle_type'] for _ in starts_at]))
+
+        diag_summary['diagnostic_type'] = pd.Series(list(itertools.chain.from_iterable(
+            [diagnostic_available['cycle_type'] for _ in starts_at])))
```

Comment:
You can create end-to-end joined copies of lists using multiplication e. g. [1, 2, 3] * 2 = [1, 2, 3, 1, 2, 3].

I think here it would look like list(diagnostic_available['cycle_type']) * len(starts_at).

Also not sure if the cast to pd.Series is necessary, but maybe I'm missing something.

Reply:
Thanks for the tip! The casting to a series is necessary for cases when the diagnostic cycles have not completed and there is a length mis-match between the dataframe and the list. Generally, I would not do this, but in this case I think that the desired behavior is to take the length of the summary dataframe no matter what so truncating the list to the length of the dataframe is justified.